### PR TITLE
Header

### LINKED ISSUE #68: [Task] Let user log in with their wallet
I think this functionality is just hidden at the moment and hopefully easy to add back in. We will need users to log in with their wallet so that we can track their list state across sessions and devices.

I would add the top navigation bar with the wallet connect feature and nothing else. Try to make it match the style of the site.

### PR DESCRIPTION
**Pull request by 0xChijioke**
## Description

Restored wallet connection functionality to enable user wallet login to close #68 

Adjusted design elements for consistency and removed redundant code.

### CODE REVIEW

**Review by escottalexander — CHANGES_REQUESTED** [review]
Code changes look good. Nice work cleaning things up.

Only issue is the nav bar box shadow making the border disappear on the impact vector selection menu. Adjusting the z-index seems to fix it though I didn't test it much to make sure it didn't cause more issues.
image

**Comment by 0xChijioke** [comment]
I've made the changes mentioned. 

I added a tiny bottom margin to the nav bar, and it has sorted out the issue with the box shadow. 

What do you think?
image

File: packages/nextjs/components/Header.tsx [hunk 1]
```diff
@@ -8,12 +8,10 @@ import { FaucetButton, RainbowKitCustomConnectButton } from "~~/components/scaff
  */
 export const Header = () => {
   return (
-    <div className="sticky lg:static top-0 navbar bg-base-100 min-h-0 flex-shrink-0 z-20 shadow-md shadow-secondary px-0 sm:px-2">
-      <div className="md:navbar-end text-right items-center lg:w-1/2">
+    <div className="sticky lg:static top-0 navbar bg-base-100 min-h-0 flex-shrink-0 z-20 mb-2 shadow-md shadow-secondary px-0 sm:px-2">
```
Comment (escottalexander) [inline packages/nextjs/components/Header.tsx:11]: With the extra margin it makes it look like the navbar elements are not centered vertically. How about we remove the shadow completely? Remove the following properties: `mb-2 shadow-md shadow-secondary`. I played around with changing the color of the shadow but I didn't like the result.

**Review by escottalexander — CHANGES_REQUESTED** [review]
Made a different suggestion.

File: packages/nextjs/components/Header.tsx (hunk 1 above)
Comment (0xChijioke) [reply packages/nextjs/components/Header.tsx]: > With the extra margin it makes it look like the navbar elements are not centered vertically. How about we remove the shadow completely? Remove the following properties: `mb-2 shadow-md shadow-secondary`. I played around with changing the color of the shadow but I didn't like the result.

done.

**Review by escottalexander — CHANGES_REQUESTED** [review]
Sorry I didn't notice this sooner. Check out what happens when yu press the connect wallet button:
image

**Comment by 0xChijioke** [comment]
> Sorry I didn't notice this sooner. Check out what happens when you press the connect wallet button: image

Thanks for catching that! 

I thought it was already sorted.

Should be good now, but trying to test it in a different envirenomet.

**Review by escottalexander — APPROVED** [review]
This is looking great! Thanks for adding this functionality back in. So much we can do now that a user can log in.

**Pull request merged**